tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,146 @@
+
+        :param setup: fixture
+        :param expected: output expected from detach_bundle()
+
+        """
+        device_data[0]['imageBundle'] = None
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+        mock__get_device.return_value = cv_data
+        if not expected:
+            # removing node_id for failure test-case
+            image_bundle['id'] = None
+
+        result = cv_tools.detach_bundle(user_inventory=user_topology)
+        assert result[0].success == expected
+        assert result[0].changed == expected
+
+        # resetting imageBundle id
+        image_bundle['id'] = 'imagebundle_1658329041200536707'
+        device_data[0]['imageBundle'] = 'EOS-4.25.4M'
+
+    def test_detach_bundle_curent_image_none(self, setup):
+        """
+        Test when current_image_bundle is none
+        :param setup: fixture
+
+        if current_image_bundle, returns empty list
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+
+        mock__get_device.return_value = None
+
+        result = cv_tools.detach_bundle(user_inventory=user_topology)
+        assert not result
+
+    def test_detach_bundle_device_image_none(self, setup):
+        """
+        Test when device.image_bundle is not None
+        :param setup: fixture
+
+        if device.image_bundle is not None, result_data have default values
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+
+        mock__get_device.return_value = cv_data
+
+        result = cv_tools.detach_bundle(user_inventory=user_topology)
+        assert result[0].success == False
+        assert result[0].changed == False
+
+    def test_detach_bundle_cvp_api_error(self, setup):
+        """
+        Test for CvpApiError.
+        :param setup: fixture
+
+        if image_bundle['id'] = 'error_id', raises CvpApiError
+        """
+        device_data[0]['imageBundle'] = None
+        user_topology = DeviceInventory(data=device_data)
+        mock_ansible_module, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+        mock__get_device.return_value = cv_data
+        image_bundle['id'] = 'error_id'
+
+        with pytest.raises(SystemExit) as pytest_error:
+            _ = cv_tools.detach_bundle(user_inventory=user_topology)
+        assert pytest_error.value.code == 1
+        expected_call = [call.fail_json(msg='Error removing bundle from device tp-avd-leaf2 : Image bundle ID is not valid')]
```

Comment:
```suggestion
        expected_call = [call.fail_json(msg='Error removing bundle from device tp-avd-leaf2: Image bundle ID is not valid')]
```
Also change this line: https://github.com/aristanetworks/ansible-cvp/blob/f8b8e40c44dd44299b38a73409efc803281bcef9/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py#L1696
to 
```suggestion
self.__ansible.fail_json(
                                msg="Error removing bundle from device "
                                + device.fqdn
                                + ": "
                                + str(catch_error)
                            )
```